sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobUserAgentModificationPolicy.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.implementation.util;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This policy modifies the blob user agent string for clients created in packages that are dependencies of blob.
+ * It transforms a User Agent String as follows
+ * UAbefore: "azsdk-java-azure-storage-blob/12.11.0-beta.2 (11.0.6; Windows 10; 10.0)"
+ * UAafter: "azsdk-java-azure-storage-blob/12.11.0-beta.2 azsdk-java-azure-storage-blob-batch/12.8.0-beta.2 (11.0.6; Windows 10; 10.0) "
+ */
+public class BlobUserAgentModificationPolicy implements HttpPipelinePolicy {
+
+    private final String clientName;
+    private final String clientVersion;
+
+    private static final String USER_AGENT = "User-Agent";
+    private static final String REGEX = "azsdk-java-azure-storage-blob/\\d+\\.\\d+\\.\\d+[-beta\\.\\d+]*(.)*";
```

Comment:
For now we may want to tighten this regex:

```suggestion
    private static final String REGEX = "azsdk-java-azure-storage-blob/12\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})?(.)*";
```

Major version is always 12, minor, patch, and beta versions are almost always 1 or 2 digits, and beta identifier is optional.